src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -66,17 +69,24 @@ public TestRequestManager()
                 TestPlatformFactory.GetTestPlatform(),
                 TestLoggerManager.Instance,
                 TestRunResultAggregator.Instance,
-                TestPlatformEventSource.Instance)
+                TestPlatformEventSource.Instance,
+                MetricsPublisherFactory.GetMetricsPublisher(IsTelemetryOptedIn(), CommandLineOptions.Instance.IsDesignMode))
+        {
+        }
+
+        public void Dispose()
         {
+            this.metricsPublisher.Result.Dispose();
```

Comment:
Add the regular pattern of handling the Dispose with one more method `Dispose(bool disposing)`

Reply:
Should I add destructor as well in that case?